lib/Parse/Lexer.cpp

```diff
@@ -1163,9 +1174,13 @@ unsigned Lexer::lexCharacter(const char *&CurPtr, char StopQuote,
     }
     // Move the pointer back to EOF.
     --CurPtr;
-    LLVM_FALLTHROUGH;
+    if (EmitDiagnostics)
+      diagnose(CurPtr-1, diag::lex_unterminated_string);
+    return ~1U;
   case '\n':  // String literals cannot have \n or \r in them.
   case '\r':
+    if (Modifiers & StringLiteralMultiline) // ... unless they are mutli-line
```

Comment:
Nit, typo. (Also, decide if you want to write "multi-line" or "multiline" throughout, and camel-case accordingly in the code.)